compiler/noirc_evaluator/src/ssa/ir/instruction/binary.rs

```diff
@@ -15,11 +15,11 @@ use super::{
 #[derive(Debug, PartialEq, Eq, Hash, Copy, Clone, Serialize, Deserialize)]
 pub(crate) enum BinaryOp {
     /// Addition of lhs + rhs.
-    Add,
+    Add { unchecked: bool },
     /// Subtraction of lhs - rhs.
-    Sub,
+    Sub { unchecked: bool },
     /// Multiplication of lhs * rhs.
-    Mul,
+    Mul { unchecked: bool },
     /// Division of lhs / rhs.
     Div,
```

Comment:
We can leave out of scope for this PR but just noting that we can have this unchecked by removing the zero denominator check.